packages/abi-coder/src/json-abi.ts

```diff
@@ -3,203 +3,44 @@
-  readonly inputs?: ReadonlyArray<JsonAbiFragmentType>;
-  readonly outputs?: ReadonlyArray<JsonAbiFragmentType>;
-  readonly attributes?: ReadonlyArray<JsonAbiFunctionAttributeType>;
-}
-
-export interface JsonAbiLogFragment {
-  readonly logId: number;
-  readonly loggedType: JsonFlatAbiFragmentArgumentType;
-  readonly abiFragmentType?: ReadonlyArray<JsonAbiFragmentType>;
+export interface JsonAbi {
+  readonly types: readonly JsonAbiType[];
+  readonly loggedTypes: readonly JsonAbiLoggedType[];
+  readonly functions: readonly JsonAbiFunction[];
+  readonly configurables: readonly JsonAbiConfigurable[];
 }
 
-export interface JsonFlatAbiFragmentType {
+export interface JsonAbiType {
   readonly typeId: number;
   readonly type: string;
-  readonly name?: string;
-  readonly components?: ReadonlyArray<JsonFlatAbiFragmentArgumentType> | null;
-  readonly typeParameters?: ReadonlyArray<number> | null;
+  readonly components: readonly JsonAbiArgument[] | null;
+  readonly typeParameters: readonly number[] | null;
 }
-
-export interface JsonFlatAbiFragmentLoggedType {
-  readonly logId: number;
-  readonly loggedType: JsonFlatAbiFragmentArgumentType;
-}
-
-export interface JsonFlatAbiFragmentArgumentType {
+export interface JsonAbiArgument {
   readonly type: number;
-  readonly name?: string;
-  readonly typeArguments?: ReadonlyArray<JsonFlatAbiFragmentArgumentType> | null;
+  readonly name: string;
+  readonly typeArguments: readonly JsonAbiArgument[] | null;
 }
 
-export interface JsonFlatAbiFragmentConfigurable {
-  name: string;
-  configurableType: JsonFlatAbiFragmentArgumentType;
-  offset: number;
+export interface JsonAbiLoggedType {
+  readonly logId: number;
+  readonly loggedType: JsonAbiArgument;
 }
 
-export interface JsonFlatAbiFragmentFunction {
+export interface JsonAbiFunction {
   readonly name: string;
-  readonly inputs?: ReadonlyArray<JsonFlatAbiFragmentArgumentType>;
-  readonly output?: Readonly<JsonFlatAbiFragmentArgumentType>;
-  readonly attributes?: ReadonlyArray<JsonAbiFunctionAttributeType> | null;
+  readonly inputs: readonly JsonAbiArgument[];
+  readonly output: JsonAbiArgument;
+  readonly attributes: readonly JsonAbiFunctionAttribute[] | null;
 }
 
-export interface JsonFlatAbi {
-  readonly types: ReadonlyArray<JsonFlatAbiFragmentType>;
-  readonly loggedTypes: ReadonlyArray<JsonFlatAbiFragmentLoggedType>;
-  readonly functions: ReadonlyArray<JsonFlatAbiFragmentFunction>;
-  readonly configurables: ReadonlyArray<JsonFlatAbiFragmentConfigurable>;
-}
-
-export interface ConfigurableFragment extends JsonFlatAbiFragmentConfigurable {
-  fragmentType: JsonAbiFragmentType;
+export interface JsonAbiFunctionAttribute {
+  readonly name: string;
+  readonly arguments: ReadonlyArray<string>;
 }
 
-export const isFlatJsonAbi = (jsonAbi: JsonAbi): jsonAbi is JsonFlatAbi => !Array.isArray(jsonAbi);
-/**
- * A JSON ABI object
- */
-export type JsonAbi = ReadonlyArray<JsonAbiFragment> | JsonFlatAbi;
-
-export class ABI {
```

Comment:
This class served the purpose of bridging `JsonFlatAbi` into the old format, so it's not needed anymore.